src/compiler/checker.ts

```diff
@@ -11205,7 +11205,7 @@ namespace ts {
                         seen = c === node;
                     }
                 });
-                if (subsequentNode) {
+                if (subsequentNode && subsequentNode.pos === node.end) {

```

Comment:
Can you add a comment that you need this test because garbage declarations may be introduced between overloads and cause issues with overloads.